### PR TITLE
Some changes such that we are able to use the analyzer in our system

### PR DESCRIPTION
**Pull request by cbumae**

### CODE REVIEW

File: analyzers.py [hunk 1]
```diff
@@ -130,45 +129,52 @@ class CobsXDecoder(HighLevelAnalyzer):
...
+                    retval = AnalyzerFrame(
+                        "Error",
+                        self.frame_start_time,
+                        self.frame_end_time,
+                        {
+                            "error": str(de),
+                        }
+                    )
+            self.received = []
+            self.frame_start_time = frame.start_time
```
Comment (jonathangjertsen) [inline analyzers.py:173]: Should set `self.frame_end_time = frame.end_time` here too, in case another null byte is sent (i.e., empty frame)

**Comment by jonathangjertsen** [comment]
Thanks for the contribution!

I'm not sure about the feature to add prefix bytes since it doesn't seem to be part of the protocol. I would prefer not to do any additional parsing of the frames, so it is more easy to adapt to different usages.

File: analyzers.py (hunk 1 above)
Comment (cbumae) [reply analyzers.py:173]: True, fixed it.

**Comment by cbumae** [comment]
> I'm not sure about the feature to add prefix bytes since it doesn't seem to be part of the protocol. I would prefer not to do any additional parsing of the frames, so it is more easy to adapt to different usages.

I agree that it is not part of the protocol. But I have seen this used in several companies already.

I tried to make it as flexible as possible: so there is no additional parsing (the prefix bytes are just discarded). Also this is configurable so by leaving the setting at 0 the behavior is unchanged compared to now.

So shall I remove it from the pull request again?

**Comment by jonathangjertsen** [comment]
Yes, please remove it. Hopefully Saleae implements [HLAs as inputs to HLAs](https://ideas.saleae.com/b/feature-requests/hla-support-input-from-another-hla/), then it would be quite easy to make other HLAs on top of it to support custom protocol stakcs.

**Comment by cbumae** [comment]
I made a new pull request here: https://github.com/jonathangjertsen/cobs-decoder-hla/pull/2

**Pull request closed without merging**